src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenClosureLambdaTests.cs

```diff
@@ -5377,5 +5377,46 @@ .maxstack  3
   IL_0059:  ret
 }");
         }
+
+        [WorkItem(44720, "https://github.com/dotnet/roslyn/issues/44720")]
+        [Fact]
+        public void LambdaInsideGenericLocalFunctionInsideLoop()
```

Comment:
>LambdaInsideGenericLocalFunctionInsideLoop [](start = 20, length = 42)

Given this test scenario. Could you please explain what was going wrong? What types and methods are involved and how the change fixes the problem?

Reply:
Added the test description.